### DISCUSSION TITLE
Bleeding edge binaries for Raspberry Pi Zero avaliable from github

### DISCUSSION THREAD

**Discussion by jcard0na** [cardonabits/haxo-hw#74] [Announcements]
Hello!

Thanks to the [great work of Peter Moz](https://github.com/cardonabits/haxo-rs/pull/33), we now have github compile the latest `haxo001` executable for us.  (_At the moment, this is only for Raspberry Pi Zero and Pi Zero 2_)

You can fetch the binary from the [actions page](https://github.com/cardonabits/haxo-rs/actions) in `haxo-rs`.  Select the latest successful run, and download the file form the `Artifacts` section.

![image](https://github.com/cardonabits/haxo-hw/assets/676181/99a3244e-f742-478b-b866-59bdcbbadc9f)

To install, replace the `haxo001` binary you get from github with the one on your `haxophone`, and you should be good to go (but maybe keep a copy of the original file to be safe...)

This is all very recent and barely tested, so only try it if you are feeling adventurous!

Cheers,
Javier

**Comment by Ekolide** [comment]
Hi! Just tried the binary on my Haxophone. Worth noting is that only users that are logged into GitHub and have read access to the repository can download artifacts.

After manually adding `--transpose 0` to my `haxo.service` The Low B fingering worked just fine. After changing the transposition a couple of times the interface started to click. I should note that the beeps of the transposition modes does not send over MIDI USB. Neither does the instrument changing mode or the startup sound. 

Next I checked the MIDI event logs, and there was no unnecessary stream of MIDI CC7 messages. Hooray! Latency with my laptop is still too high to be fun to use, and I can't intuitively say if there's a positive difference from the update. There are some redundant messages at the very lowest breath values if you slowly decrease your breath down to 0, causing several Note On and Note Off messages being sent.

On that topic, I checked the output of my AKAI EWI USB for fun, and it generally doesn't share this behavior. Rather, it requires quite a bit more air pressure compared to the Haxophone to start sending MIDI information, and it maxes out at a lower air pressure than the Haxophone. This makes the AKAI less sensitive in the lower ranges. So this might not be worth thinking too much about in Haxophone world.

Lots of fun!

**Comment by jcard0na** [comment]
Thanks for trying this out and providing feedback.

> After changing the transposition a couple of times the interface started to click. 

What do you mean by this?  

> I should note that the beeps of the transposition modes does not send over MIDI USB. Neither does the instrument changing mode or the startup sound.

Yes, that is how it is implemented at the moment.  The main use case for USB MIDI was transcribing, so any control sounds (startup chime, confirmation sounds, etc.) are not sent via MIDI.  If you think they should, please open an issue.

Cheers, and thanks again!

**Reply by jcard0na** [reply]
> > If you think they should, please open an issue.

Oh, never mind! 😄 https://github.com/cardonabits/haxo-rs/issues/37

**Comment by Ekolide** [comment]
So what I mean about the transposition intrerface. It simply took some getting used to, with inhaling to get to the mode, exhaling to pick a note. And even though I've read the implementation details in the issue page and pull request I still needed to sit with a piano a couple of times to get a grip on how the transposition is "centered" around the mid C fingering.

Cheers!

**Reply by jcard0na** [reply]
Ah, yes, "click" as in "to be understood, to become clear" not as in "to make short, sharp sounds" (For a moment I thought you might have uncovered a problem).
Thanks for clarifying.
